### ISSUE TITLE
Server-side performance issue with small I/O

### ISSUE THREAD

**Issue by wangvsa** [hpc-io/pdc#190]
This issue documents the performance challenges observed during the Montage experiments. The high server-side overhead encountered when processing small I/O requests needs to be mitigated to make PDC beneficial for Montage or AI applications.

**What does this feature solve or improve?**

The Montage results suggest that the server may not be operating at its peak efficiency. In scenarios with numerous concurrent requests, the server could potentially become a bottleneck. We may observe similar I/O patterns from AI applications as well.

**Describe the solution you'd like**

Server side algorithm for processing I/O requests can be improved.
Server-side multi-threading should also be able to improve the efficiency.

**Montage results on Perlmutter**

The Montage components execute a large number of small reads and writes. Within the tested workflow, each I/O operation amounts to approximately 3000 bytes.
The performance of PDC, with or without cache, remains similar, indicating that the majority of the time was consumed by server processing
<img width="700" alt="image" src="https://github.com/hpc-io/pdc/assets/2760209/bb89084b-f6b2-4a4c-a4a7-612943eec830">

I did some further investigations on one component, mProjExecMPI. This component executes N small writes, followed by one read, and then another M writes. I implemented optimizations, including utilizing session consistency and combining all writes into one batched call. However, the performance remains suboptimal. Especially, the single read operation takes 2 seconds, suggesting it was awaiting processing on the server side.

<img width="700" alt="image" src="https://github.com/hpc-io/pdc/assets/2760209/877ec841-230e-4b2a-8c93-302d1a8c2196">

**Comment by jeanbez** [comment]
@wangvsa could you please confirm this has solved the issue so we can close this as the PR was merged?

**Comment by wangvsa** [comment]
Yes, I just re-ran the experiments, and the performance has significantly improved. Preliminary results show that the performance of this test is now comparable to that of Lustre.
